filesystem/src/main/java/org/frankframework/filesystem/FileSystemActor.java

```diff
@@ -291,10 +291,24 @@ private F getFile(@Nonnull Message input, ParameterValueList pvl) throws FileSys
 	private F getFileAndCreateFolder(@Nonnull Message input, ParameterValueList pvl) throws FileSystemException {
 		String filenameWithFolder = determineFilename(input, pvl);
 		String folder = FilenameUtils.getFullPathNoEndSeparator(filenameWithFolder);
-		if(StringUtils.isNotBlank(folder) && isCreateFolder() && !fileSystem.folderExists(folder)) {
+
+		if (StringUtils.isNotBlank(folder)) {
+			createFolder(folder);
+
+			return fileSystem.toFile(filenameWithFolder);
+		}
+
+		// If the folder could not be determined on the filename, let the filesystem implementation determine (and create, if needed) the folder
+		F file = fileSystem.toFile(filenameWithFolder);
+		createFolder(fileSystem.getParentFolder(file));
+
+		return file;
+	}
+
+	private void createFolder(String folder) throws FileSystemException {
+		if (isCreateFolder() && !fileSystem.folderExists(folder)) {
 			fileSystem.createFolder(folder);
 		}
```

Comment:
Ik denk dat we in onze api (interface) moeten zetten wanneer deze folderExist check gedaan moet worden.
Ik weet toevallig dat oa in de `FtpFileSystem#createFolder` ook al een dergelijke check is, en dan is hij onnodig duppel-op.

Reply:
Net ook nog even naar Tim's pr gekeken, ik denk dat het handiger is hier een nieuw issue voor te maken om de performance iets te verbeteren en duidelijk te maken waar welke checks gedaan moeten worden.

Reply:
Ik heb #6810 aangemaakt hiervoor.